combat-trainer.lic

```diff
@@ -1997,7 +2025,186 @@ class TrainerProcess
     end
     DRRoom.dead_npcs.size > dead_count
   end
+  def visible_bodies
+    result = []
+    all_bodies = @constellations
+    case DRC.bput('observe sky', 'The following heavenly bodies are visible:', "That's a bit hard to do while inside")
+    when "That's a bit hard to do while inside"
+      echo('Must be outdoors to observe sky')
+      exit
+    end
+    until (line = get?) =~ /^Roundtime/i
+      result << all_bodies.find { |body| /\b#{body['name']}\b/i =~ line && line !~ /below the horizon/ }
+    end
+    result.compact.select { |data| data['circle'] <= DRStats.circle }
+  end
+  
+  def check_pools
+    pools = {
+      'lore' => 0,
+      'magic' => 0,
+      'survival' => 0,
+      'offensive combat' => 0,
+      'defensive combat' => 0,
+      'future events' => 0
+    }
+    pools_to_size = {
+      /You have no understanding of the celestial influences over/ => 0,
+      /You have a feeble understanding of the celestial influences over/ => 1,
+      /You have a weak understanding of the celestial influences over/ => 2,
+      /You have a fledgling understanding of the celestial influences over/ => 3,
+      /You have a modest understanding of the celestial influences over/ => 4,
+      /You have a decent understanding of the celestial influences over/ => 5,
+      /You have a significant understanding of the celestial influences over/ => 6,
+      /You have a potent understanding of the celestial influences over/ => 7,
+      /You have an insightful understanding of the celestial influences over/ => 8,
+      /You have a powerful understanding of the celestial influences over/ => 9,
+      /You have a complete understanding of the celestial influences over/ => 10
+    }
+    waitrt?
+    pause 1
+    fput("retreat")
+    fput("retreat")
+    DRC.bput('predict state all', 'roundtime')
```

Comment:
This is a 15 second roundtime which I would also like to remove from the astrology script itself. I would remove it from combat

Reply:
Sorry disregard